### ISSUE TITLE
New users have blank profiles

### ISSUE THREAD

**Issue by mikethicke** [MESH-Research/commons#70]
Tester (if different from issue creator): Anne
Site: MLA Commons, HC 
URL: https://mla.hcommons-dev.org/members/hctester40/
Test Account: Ferdinand Test / Any new account
Area (CORE, sites, profiles, groups, general): profiles
If issue discovered on dev, does it also exist on prod? No

**Describe the issue**

Ferdinand Test profile is missing fields; doesn't appear in list of members on directory. Any new user has blank profile, in both view and edie

**To Reproduce**
1. Enroll a new user
2. Visit their profile. It is blank.
3. Edit their profile. No fields appear.
4. Save the profile.
5. Fields now appear and profile is no longer blank.

**Expected behavior**
When a new user creates a profile, viewing the profile should show academic interests, but be otherwise blank. When user clicks edit, they should see fields to update.

**Screenshots**
<img width="960" alt="dianatestWP" src="https://user-images.githubusercontent.com/26980498/129387079-7c056999-7755-497c-b202-20fece8e2c5b.png">

**Comment by mikethicke** [comment]
TLDR: I believe the issue is linked to the enrollment problem we've been seeing elsewhere, and fixing that problem would fix this problem. However, we can fix this problem without fixing that problem, and probably should.

Longer version:

To loop through profile fields for view or edit, the [_hcmp_do_with_field_in_loop](https://github.com/MESH-Research/hc-member-profiles/blob/e94b5698c29aba4e66de4753613cf2b775d0b67b/includes/functions.php#L287) function is called:

```
function _hcmp_do_with_field_in_loop( $field_name = '', $callback ) {
	$args = [
		'hide_empty_fields' => false, // Some custom field types are "empty" by design e.g. 'CORE Deposits'.
	];

	if ( bp_has_profile( $args ) ) {
		while ( bp_profile_groups() ) {
			bp_the_profile_group();

			if ( bp_profile_group_has_fields() ) {
				while ( bp_profile_fields() ) {
					bp_the_profile_field();

					if ( bp_get_the_profile_field_name() === $field_name ) {
						return call_user_func( $callback );
					}
				}
			}
		}
	}
}
```

The inner while loop is gated by ```if ( bp_profile_group_has_fields() )```, which is unnecessary since the while loop won't run if ```bp_profile_fields()``` returns no fields to iterate over. Further, ```bp_profile_group_has_fields()``` doesn't do what it says it does. It returns true only if there are fields **with data**. ```bp_profile_fields()``` loops over all fields, regardless of whether they have data. When a new user enrolls on prod, there is one field with data: the name field, which is their WordPress username. When a new user enrolls on dev, the name field is empty, and so there are no fields with data, and so the while loop never runs, even though there are fields to iterate over.

If the ```if ( bp_profile_group_has_fields() )``` is removed, then the profile displays and can be edited as on dev just as on prod.

Since this clause is unnecessary and is causing a major issue on dev, I think we should just remove it even though the issue will probably go away when the enrollment issue is solved.
